src/app/api/oneproduct.service.ts

```diff
@@ -0,0 +1,50 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, catchError, throwError } from 'rxjs';
+import { IOneSingleProduct } from '../Interface/singleproduct';
+import { CountryService } from '../services/country.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OneProductService {
+  oneProduct!: IOneSingleProduct | undefined;
+
+  private oneProductUrl =
+    'https://ac-trainee-store-api.herokuapp.com/products/';
+
+  private oneProductSubject: BehaviorSubject<IOneSingleProduct | undefined> =
+    new BehaviorSubject(this.oneProduct);
+
+  oneProduct$: Observable<IOneSingleProduct | undefined> =
+    this.oneProductSubject.asObservable();
+
+  setOneProductSubject(newValue: IOneSingleProduct | undefined) {
+    this.oneProductSubject.next(newValue);
+  }
+
+  constructor(private http: HttpClient, private country: CountryService) {}
+
+  getSingleProductDetails(
```

Comment:
Would it be okay for you to rename this to getProductDetails?
The fact that its a only one product is stated by "Product" which is singular. If the method would return more then one product the name should contain the plural "Products".

Reply:
"getSingleProductDetails"  was renamed to "getProductDetails"